cumulusci/robotframework/utils.py

```diff
@@ -1,7 +1,48 @@
+import time
+import functools
+from selenium.common.exceptions import ElementNotInteractableException
+from selenium.common.exceptions import StaleElementReferenceException
+from selenium.common.exceptions import WebDriverException
+from SeleniumLibrary.errors import ElementNotFound
+
 def set_pdb_trace():
     import sys
     import pdb
 
     for attr in ("stdin", "stdout", "stderr"):
         setattr(sys, attr, getattr(sys, "__%s__" % attr))
     pdb.set_trace()
+
+ALWAYS_RETRY_EXCEPTIONS=(
+    ElementNotFound,
+    ElementNotInteractableException,
+    StaleElementReferenceException,
+)
+
+def selenium_retry(func):
+    @functools.wraps(func)
+    def deco_selenium_retry(*args, **kwargs):
+        retry = False
+        try:
+            return func(*args, **kwargs)
+        except ALWAYS_RETRY_EXCEPTIONS as e:
+            retry = True
+        except WebDriverException as e:
+            if "Other element would receive the click" in e.msg:
+                retry = True
+        except Exception as e:
+            pass
+        if retry:
+            args[0].builtin.log(
+                "Retrying call to method {}".format(func.__name__), level="WARN" 
+            )
+            time.sleep(2)
+            return func(*args, **kwargs) 
+        else:
+            args[0].selenium.capture_page_screenshot()
```

Comment:
what is `args[0]`? either assign to a variable or give me a comment. its used multiple times.

Reply:
Just make `self` the first arg of deco_selenium_retry, and call func(self, *args, **kwargs). Then you can use self here.